### PR TITLE
Make RNN layers accept `in => out`

### PR DESCRIPTION
**Pull request by mcabbott**
This does #1658 for all the RNN layers.

### CODE REVIEW

File: src/layers/recurrent.jl [hunk 1]
```diff
@@ -187,7 +187,7 @@ struct LSTMCell{A,V,S}
   state0::S
 end
 
-function LSTMCell(in::Integer, out::Integer;
+function LSTMCell((in, out)::Pair{<:Integer, <:Integer};
```
Comment (ToucheSir) [inline src/layers/recurrent.jl:190]: Why is the additional eltype constraint required here in particular?

Comment (mcabbott) [reply src/layers/recurrent.jl:190]: This signature just happens to already be written out on different lines.

Comment (ToucheSir) [reply src/layers/recurrent.jl:190]: Sorry, not sure I follow. Is the plan to update `RNNCell` and `GRU(v3)Cell` to match?

Comment (mcabbott) [reply src/layers/recurrent.jl]: No. I typed this one this way. Then I thought it doesn't really matter, and isn't worth the clutter of changing the others to be multi-line definitions. 

Now fixed, they all agree. 

(Previously, `GRUCell(in, out;` took Any, Any, btw.)

**Review by ToucheSir — APPROVED** [review]
LGTM!

**Pull request merged**